src/state_manager/mod.rs

```diff
@@ -1271,12 +1269,33 @@ where
             Some(child)
         });
 
+        self.validate_stream(epochs, tipsets)
+    }
+
+    pub fn validate_stream<F>(
```

Comment:
I don't understand why this new function is necessary - could you help me understand?